Arrays Calculation application

Geirby opened the pull request:
Implementation of arrays calculation

Geirby committed:
Arrays Calculation application

Dauren-Delmukhambetov commented on src/main/java/com/company/Main.java:
```diff
@@ -1,32 +1,134 @@
-
-        System.out.println("Your result  " + String.format("%.2f%n", result));
+        System.out.println("Pls enter initial value of 1st array");
+        var inValueForFirstArray = in.nextInt();
```
You can declare this var as `final` since its user-input data and shouldn't be changed

Dauren-Delmukhambetov commented on src/main/java/com/company/Main.java:
```diff
@@ -1,32 +1,134 @@
+        System.out.println("Pls enter size of arrays");
+        var sizeOfArrays = in.nextInt();
+        System.out.println("Pls enter step of 1st array");
+        var stepForFirstArray = in.nextInt();
```
Since array eventually has a `double` type it would be better to allow users to enter double values

Dauren-Delmukhambetov commented on src/main/java/com/company/Main.java:
```diff
@@ -1,32 +1,134 @@
+        System.out.println("Pls enter step of 1st array");
+        var stepForFirstArray = in.nextInt();
+        System.out.println("Pls enter increment operator of 1st array (+),(*),(-) or (/)");
+        String operatorForFirstArray = in.next();
```
For the sake of consistency, it would be better to use `var` here too

Dauren-Delmukhambetov commented on src/main/java/com/company/Main.java:
```diff
@@ -1,32 +1,134 @@
+        System.out.println("Pls enter step of 2nd array");
+        var stepForSecondArray = in.nextInt();
+        System.out.println("Pls enter increment operator of 2nd array (+),(*),(-),(^) or (/)");
+        String operatorForSecondArray = in.next();
```
It would be more readable to use `nextLine()` to get String value

Dauren-Delmukhambetov commented on src/main/java/com/company/Main.java:
```diff
@@ -1,32 +1,134 @@
+        var sizeOfArrays = in.nextInt();
+        System.out.println("Pls enter step of 1st array");
+        var stepForFirstArray = in.nextInt();
+        System.out.println("Pls enter increment operator of 1st array (+),(*),(-) or (/)");
```
You missed `^` operation here

Dauren-Delmukhambetov commented on src/main/java/com/company/Main.java:
```diff
@@ -1,32 +1,134 @@
+        System.out.println("Pls enter increment operator of 2nd array (+),(*),(-),(^) or (/)");
+        String operatorForSecondArray = in.next();
+        System.out.println("Pls enter operator to calculate the third array (+),(*),(-),(^) or (/)");
+        String GeneralOperator = in.next();
```
Try to follow [Java Naming Convention](https://www.oracle.com/java/technologies/javase/codeconventions-namingconventions.html)
Here, the `generalOperator` should be in camel case

Dauren-Delmukhambetov commented on src/main/java/com/company/Main.java:
```diff
@@ -1,32 +1,134 @@
+        String operatorForSecondArray = in.next();
+        System.out.println("Pls enter operator to calculate the third array (+),(*),(-),(^) or (/)");
+        String GeneralOperator = in.next();
+        boolean x = false;
```
Try to give descriptive names to variables 
Here it's not clear what your intention was

Dauren-Delmukhambetov commented on src/main/java/com/company/Main.java:
```diff
@@ -1,32 +1,134 @@
-    }
 
+        double[] array1 = new double[sizeOfArrays];
+        array1[0] = inValueForFirstArray;
```
Misleading naming, `initValueForFirstArray`

Dauren-Delmukhambetov commented on src/main/java/com/company/Main.java:
```diff
@@ -1,32 +1,134 @@
+        for (int i = 1; i < sizeOfArrays; i++) {
+            switch (operatorForFirstArray) {
+                case "+":
+                    array1[i] = array1[i - 1] + stepForFirstArray;
```
Can we write this in the next way?
```suggestion
                    array1[i] = array1[i--] + stepForFirstArray;
```

Dauren-Delmukhambetov commented on src/main/java/com/company/Main.java:
```diff
@@ -1,32 +1,134 @@
+            if (x) {
+                System.out.println("Error: Wrong operator for first array");
+                break;
+
+
```
Unneeded blanc lines

Dauren-Delmukhambetov commented on src/main/java/com/company/Main.java:
```diff
@@ -1,32 +1,134 @@
+
+            }
+        }
+        String array1String = Arrays.toString(array1);
```
Good usage of `Arrays` utility class

Dauren-Delmukhambetov commented on src/main/java/com/company/Main.java:
```diff
@@ -1,32 +1,134 @@
+
+
+        double[] array2 = new double[sizeOfArrays];
+        array2[0] = inValueForSecondArray;
```
The same as about naming above

Dauren-Delmukhambetov commented on src/main/java/com/company/Main.java:
```diff
@@ -1,32 +1,134 @@
+
+        double[] array2 = new double[sizeOfArrays];
+        array2[0] = inValueForSecondArray;
+        for (int i = 1; i < array1.length; i++) {
```
It would be better to refer to the size if second array since you are operating on it

Dauren-Delmukhambetov commented on src/main/java/com/company/Main.java:
```diff
@@ -1,32 +1,134 @@
+                    array2[i] = Math.pow(array1[i - 1], stepForSecondArray);
+                    break;
+                default:
+                    i = sizeOfArrays;
```
Modifying counter inside of the loop can be considered as bad practice, try to avoid this

Dauren-Delmukhambetov commented on src/main/java/com/company/Main.java:
```diff
@@ -1,32 +1,134 @@
+
+        double[] array3 = new double[sizeOfArrays];
+        switch (GeneralOperator) {
+            case "+":
+                for (int i = 0; i < array1.length; i++) {
+                    array3[i] = array1[i] + array2[i];
+                }
+                break;
+            case "-":
+                for (int i = 0; i < array1.length; i++) {
+                    array3[i] = array1[i] - array2[i];
+                }
+                break;
+            case "*":
+                for (int i = 0; i < array1.length; i++) {
+                    array3[i] = array1[i] * array2[i];
+                }
+                break;
+            case "/":
+                for (int i = 0; i < array1.length; i++) {
+                    array3[i] = array1[i] / array2[i];
+                }
+                break;
+            case "^":
+                for (int i = 0; i < array1.length; i++) {
+                    array3[i] = Math.pow(array1[i], array2[i]);
+                }
```
Could we re-write this with one usage of the `for` loop, how do you think?

Dauren-Delmukhambetov reviewed (changes requested):
Overall, the calculator works, which is great. 
I left comments mostly about code style, we will discuss them during the today's sync

Geirby committed:
Arrays Calculation application (
correction of comments)

Dauren-Delmukhambetov commented:
Reported our agreement [here](https://github.com/Dauren-Delmukhambetov/bookish-dollop/issues/6#issuecomment-895999468)

Geirby committed:
Use methods for operation

Geirby committed:
Added check for the number of array elements

Dauren-Delmukhambetov commented on src/main/java/com/company/Main.java:
```diff
@@ -1,32 +1,92 @@
+        System.out.println("initial value, size of array, step of array, increment operator((+),(*),(-),(^) or (/))");
+        var arrayParametersForFirstArray = in.nextLine();
+        System.out.println("Pls enter through space parameters for second Array: ");
+        System.out.println("initial value, size of array, step of array, increment operator((+),(*),(-),(^) or (/))");
```
You don't need array size in the second input, you can use the number from the first input

Dauren-Delmukhambetov commented on src/main/java/com/company/Main.java:
```diff
@@ -1,32 +1,92 @@
+        return array;
+    }
+
+    public static double[] CalculateArrays(double[] arrayFirst, double[] arraySecond, String operator) {
```
For method names we also use camel case

Geirby committed:
Final version

Geirby committed:
Final version. Small changes

Dauren-Delmukhambetov commented on src/main/java/com/company/Main.java:
```diff
@@ -1,32 +1,97 @@
+        double[] array = new double[sizeOfArray];
+        array[0] = initialValue;
+        for (int i = 1; i < array.length; i++) {
+            array[i] = calculateNumber(initialValue, stepOfArray, incrementOperator);
```
With this method call, you will always have the same value in your array. 
For each element of the array, you are just performing operations on the initial value again and again.

Dauren-Delmukhambetov commented:
@Geirby, the program doesn't calculate correctly. Please, check it out and fix

Geirby committed:
Final version.Changes:
1) Now user input size of array only 1 time for first array
2) Edited "calculateArrays" method
